app.js

```diff
@@ -19,3 +19,54 @@ closeMenu.addEventListener('click', close);
 portfolio.addEventListener('click', close);
 aboutMe.addEventListener('click', close);
 contactMe.addEventListener('click', close);
+
+// form validation
+
+const form = document.querySelector('#form');
+const emailText = document.querySelector('input[type="email"]');
+
+form.addEventListener('submit', (event) => {
+  if (emailText.value !== emailText.value.toLowerCase()) {
+    event.preventDefault();
+
+    // Remove previous error messages
+    const previousErrorMessage = form.querySelector('.error-message');
+    if (previousErrorMessage) {
+      form.removeChild(previousErrorMessage);
+    }
+
+    const errorMessage = document.createElement('p');
+    const formText = document.querySelector('#form textarea');
+    errorMessage.textContent = 'Please enter your email in lower case!';
+    errorMessage.classList.add('error-message');
+
+    formText.insertAdjacentElement('afterend', errorMessage);
+  }
+});
+
+// Store form data in localStorage
+// Select the form element
+const form1 = document.querySelector('#form');
+
+// Listen for the form submit event
+form1.addEventListener('submit', (event) => {
+  event.preventDefault(); // Prevent default form submission behavior
+
+  // Get form input values
+  const name = document.querySelector('#name').value;
+  const email = document.querySelector('#mail').value;
+  const message = document.querySelector('#msg').value;
+
+  // Create a JavaScript object to hold the form data
+  const form1Data = {
+    name,
+    email,
+    message,
+  };
+
+  // Store the form data in local-Storage
+  localStorage.setItem('form1Data', JSON.stringify(form1Data));
+
+  // Reset the form
+  form.reset();
+});
```

Comment:
- [ ] Great job implementing localStorage but as part of the requirements, you are supposed to store the data when the user changes the content of any input field. ill recommend adding an event listener on the input fields and save the data every time there is a change